src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -287,11 +287,10 @@ private void AppendOneNativeArgument(ExecutionContext context, CommandParameterI
                         if (NeedQuotes(arg))
                         {
                             _arguments.Append('"');
-                            AddToArgumentList(parameter, arg);
+                            PossiblyGlobArg(arg, parameter, usedQuotes: false);
```

Comment:
Since everything is going to be possibly globbed, we may get some additional globbing (such as expanded wildcards and variable resolution). That _may_ be a good thing, but it may be more aggressive than we want. should we be checking explicitly for the `~` before passing it to the globber?

Reply:
I think the lack of globbing was a bug previously as something like:

```powershell
1 > "hello world.txt"
ls hello` world.txt
ls hello` world*
```

The first `ls` works as expected as the native command parameter binder recognizes it needs quotes due to the escaped whitespace.  The second case doesn't do globbing which it should and this PR also fixes that.